functions/venue.js

```diff
@@ -476,23 +476,36 @@ exports.createVenue_v2 = functions.https.onCall(async (data, context) => {
 
 exports.upsertRoom = functions.https.onCall(async (data, context) => {
   checkAuth(context);
+
+  const result = {
+    provided: data,
```

Comment:
`provided` used since `data` is anything and overusing `data` is like overusing `this` everywhere: provides no additional context since everything at the end is data.

Reply:
> Do you need this field?

Yes https://github.com/sparkletown/sparkle/pull/1950#discussion_r684131622
That's an already existing data that serves as fallback

Reply:
> provided used since data is anything and overusing data is like overusing this everywhere: provides no additional context since everything at the end is data.

I agree but you are not to decide these things on your own, all of us should be notified about changes like this one and we should all discuss it before actually going with it.

I'm constantly complaining and whining about these things because the code was a complete mess a couple of months ago. It was written by 10 different people and everyone had his own style and standards. It was like 10 different styles scattered all across the project and components.

Reply:
If you change `provided` to something more specific I think this would be good to go.

Reply:
> I agree but you are not to decide these things on your own, all of us should be notified about changes like this one and we should all discuss it before actually going with it.

PR review process is a discussion.

Do you think there should be a different place for it?
And if so, what is the threshold upon which discussion is being transferred to everyone and not constrained to a PR review (ideally visible to all of @sparkletown/engineering when a request for review is made).


> If you change provided to something more specific I think this would be good to go.

I'm open to suggestions. I'm at this moment not able to find adequate name that potentially can set a precedent for other functions like `upsertRoom`.

Is `input` specific? Something else like `options`?

Reply:
>PR review process is a discussion.

Yes, PR discussion is in progress but I'm the only one reviewing it and the only one actually seeing this new code style. Mike and Viktoria will see this in a week and wonder why it's different and not like the rest of the code. We shouldn't be doing 'shadow' changes like this.

>Do you think there should be a different place for it?
>And if so, what is the threshold upon which discussion is being transferred to everyone and not constrained to a PR >review (ideally visible to all of @sparkletown/engineering when a request for review is made).

When introducing a new code style, it should be discussed with the team in Slack. Especially with the senior devs. They should agree on the new change.

Reply:
> Yes, PR discussion is in progress but I'm the only one reviewing it and the only one actually seeing this new code style.

There is no limitation to whomever can see this PR discussion, even after a PR is closed/merged/etc.

> Mike and Viktoria will see this in a week and wonder why it's different and not like the rest of the code. 

The discussion is here, a simple @ to them will suffice, or share a link to them. PR can be resolved after they have reviewed it.

> We shouldn't be doing 'shadow' changes like this.

Please define what "'shadow change" means. This discussion isn't only visible now and care has been taken to detail changes as they could be read by anyone currently and in the future for historical reference.


> When introducing a new code style, it should be discussed with the team in Slack. Especially with the senior devs. They should agree on the new change.

And this shouldn't stop PR from being created with any code style change. Discussing code style changes before there is some PR makes for an abstract conversation, this way there is a concrete example. Anyone can share a link to this discussion on Slack before PR is merged and after it is closed.